setup.sh

```diff
@@ -43,52 +43,54 @@ declare -r CONFIGS_PATH='configs'
 
 declare -r DOCS_VIRTUAL_ENV='docs_virtual_env'
 
+function debian_pkg_check()
+{
+  dpkg-query -W --showformat='${Status}\n' "$1" 2> /dev/null | grep -c 'ok installed'
+}
+
 function check_dependencies()
 {
   local package_list=''
   local cmd=''
   local distro
-
+  readonly IS_INSTALLED_CMD=0
+  readonly DISTRO_CMD_FOR_INSTALL_PACKAGE=1
+  declare -a debian=('debian_pkg_check ' 'apt install -y')
+  declare -a arch=('pacman -Ql ' 'pacman -S')
+  declare -a gentoo=('qlist -I ' 'emerge')
+  declare -a fedora=('rpm -q ' 'dnf install -y')
+  declare -A distro_commands=(['arch']=arch ['gentoo']=gentoo ['fedora']=fedora ['debian']=debian)
+  declare -n distro_cmd
   distro=$(detect_distro '/')
+  distro_cmd="${distro_commands["$distro"]}"
 
-  if [[ "$distro" =~ 'arch' ]]; then
-    while IFS='' read -r package; do
-      installed=$(pacman -Ql "$package" &> /dev/null)
-      [[ "$?" != 0 ]] && package_list="$package $package_list"
-    done < "$DOCUMENTATION/dependencies/arch.dependencies"
-    cmd="pacman -S $package_list"
-  elif [[ "$distro" =~ 'debian' ]]; then
+  if [[ -v distro_commands["$distro"] ]]; then
     while IFS='' read -r package; do
-      installed=$(dpkg-query -W --showformat='${Status}\n' "$package" 2> /dev/null | grep -c 'ok installed')
-      [[ "$installed" -eq 0 ]] && package_list="$package $package_list"
-    done < "$DOCUMENTATION/dependencies/debian.dependencies"
-    cmd="apt install -y $package_list"
-  elif [[ "$distro" =~ 'fedora' ]]; then
-    while IFS='' read -r package; do
-      installed=$(rpm -q "$package" &> /dev/null)
+      installed=$(${distro_cmd[$IS_INSTALLED_CMD]} "$package" &> /dev/null)
       [[ "$?" -ne 0 ]] && package_list="$package $package_list"
```

Comment:
As you are checking the return of the evaluation with `[[ "$?" -ne 0 ]]`, think that the `installed` variable and the output redirection as a whole is obsolete. Consider using `eval`. This will fix both points and also allow for a safer expansion.

Reply:
okay i'll do this